qcodes/instrument_drivers/Keysight/keysightb1500/KeysightB1517A.py

```diff
@@ -1,18 +1,332 @@
+
+        self.add_parameter(name='trigger_delay',
+                           initial_value=0,
+                           unit='s',
+                           parameter_class=GroupParameter,
+                           docstring=textwrap.dedent("""
+                            Step source trigger delay time (in seconds) that
+                            is the wait time after completing a step output 
+                            setup and before sending a step output setup 
+                            completion trigger. 0 to the value of ``delay``, 
+                            with 0.1 ms resolution. 
+                            If this parameter is not set, 
+                            trigger delay will be 0.
+                            """))
+
+        self.add_parameter(name='measure_delay',
+                           initial_value=0,
+                           unit='s',
+                           vals=vals.Numbers(0, 65.535),
+                           parameter_class=GroupParameter,
+                           docstring=textwrap.dedent("""
+                           Step measurement trigger delay time (in seconds)
+                           that is the wait time after receiving a start step 
+                           measurement trigger and before starting a step 
+                           measurement. 0 to 65.535, with 0.1 ms resolution. 
+                           Numeric expression. If this parameter is not set, 
+                           measure delay will be 0.
+                           """))
+
+        self._set_sweep_delays_group = Group(
+            [self.hold,
+             self.delay,
+             self.step_delay,
+             self.trigger_delay,
+             self.measure_delay],
+            set_cmd='WT '
+                    '{hold},'
+                    '{delay},'
+                    '{step_delay},'
+                    '{trigger_delay},'
+                    '{measure_delay}',
+            get_cmd=self._get_sweep_delays(),
+            get_parser=self._get_sweep_delays_parser)
+
+        self.add_parameter(name='sweep_mode',
+                           initial_value=constants.SweepMode.LINEAR,
+                           vals=vals.Enum(*list(constants.SweepMode)),
+                           set_parser=constants.SweepMode,
+                           parameter_class=GroupParameter,
+                           docstring=textwrap.dedent("""
+                 Sweep mode. Note that Only linear sweep (mode=1 or 3) is
+                 available for the staircase sweep with pulsed bias.
+                     1: Linear sweep (single stair, start to stop.)
+                     2: Log sweep (single stair, start to stop.)
+                     3: Linear sweep (double stair, start to stop to start.)
+                     4: Log sweep (double stair, start to stop to start.)
+                                """))
+
+        self.add_parameter(name='sweep_range',
+                           initial_value=0,
```

Comment:
I am confused. Does this take the Enum values or the interger values as input.
E.g. `Auto` or `0`

Reply:
my mistake. Fixed it to take `constants.VOutputRange.AUTO` as initial value instead of 0.